scipy/optimize/__init__.py

```diff
@@ -234,7 +235,7 @@
 from ._basinhopping import basinhopping
 from ._linprog import linprog, linprog_verbose_callback
 from ._differentialevolution import differential_evolution
-
+from .least_squares import least_squares
```

Comment:
I think that ``least_squares.py`` should also start with an underscore - it's still a private module.